Source/NVActivityIndicatorView/Presenter/NVActivityIndicatorPresenter.swift

```diff
@@ -142,7 +143,10 @@ private struct NVActivityIndicatorPresenterStateAnimating: NVActivityIndicatorPr
 
 private struct NVActivityIndicatorPresenterStateWaitingToStop: NVActivityIndicatorPresenterState {
     func startAnimating(presenter: NVActivityIndicatorPresenter) {
-        // Do nothing
+        presenter.stopAnimating()
+
+        guard let activityData = presenter.data else { return }
+        presenter.startAnimating(activityData)
```

Comment:
Since it's waiting to stop, the animation isn't yet removed, would changing the state back to `.animating` suffice?

Reply:
Actually, it depends on displayTimeThreshold. 

If it is equal to zero then all should happen in the same run loop iteration (note: hide() will be called immediately when show(with: ) should be scheduled for the end of current iteration because it is called via asyncAfter). you can check the video of how it will look in this case: https://www.dropbox.com/s/jztc99mvfpfl3yk/NVActivityIndicator_PR_%23218_demo.mov?dl=0

In case when displayTimeThreshold is greater then 0 existing indicator will be dismissed immediately when new indicator will be displayed after delay (displayTimeThreshold)

Reply:
@ninjaprox What do you think about above? Could we merge it and release a new version?

Reply:
@SiarheiYurcha: I can't really see the difference from the video. I did some tries, and seems setting the state to `.animating` is sufficient in both cases of `displayTimeThreshold`.

```swift
private struct NVActivityIndicatorPresenterStateWaitingToStop: NVActivityIndicatorPresenterState {
    func startAnimating(presenter: NVActivityIndicatorPresenter) {
        presenter.state = .animating
    }

    func stopAnimating(presenter: NVActivityIndicatorPresenter) {
        presenter.hide()
        presenter.state = .stopped
    }
}
```

To explain this, the following table might help illustrate.

| | Call | State | Note |
| --- | --- | --- | --- |
| 1 | startAnimating() | .waitingToStart | |
| 2 | stopAnimating() | .stopped | |
| 3 | startAnimating() | .waitingToStart | |
| 4 | asyncAfter block | .waitingToStart | from 1 |
| 5 | startAnimating() | .animating | from 4 |
| 6 | asyncAfter block | .animating | from 2 |

The `asyncAfter` blocks are always called in the next iteration so as long as successive calls in one iteration determine the state, when the `asyncAfter` block is called, it should follow that state. What do you think?

Reply:
@ninjaprox No just change state back to .animating is not enough because we will display old ActivityData. 
E.g. if user call 
- startAnimatin(firstActivityData)
- stopAnimaing()
- startAnimating(secondActivityData)
We will display firstActivityData on the second start.

Regarding asyncAfter, I think we should not really care about this because we have a check in all asyncAfter blocks and in case if a state has been changed then the asyncAfter block will be skipped.

Reply:
Good point that I missed out. It's good to merge.